src/examples/printInterfaces.ts

```diff
@@ -14,16 +14,37 @@ export interface Bar {
 export interface Baz {
     baz: string;
 }
+
+export interface WithMethods {
+    myMethod(a: string): number;
+}
 `;
 
 const ast = createFromString(declarationFile);
 
-const interfaces: { name: string; members: string[] }[] = [];
+const interfaces: { name: string; properties: string[]; methods: string[] }[] = [];
 accept(ast, {
   [ts.SyntaxKind.InterfaceDeclaration]: (interfaceDeclaration) => {
+    const properties: string[] = [];
+    const methods: string[] = [];
+    interfaceDeclaration.members.map((member) => {
+      switch (member.kind) {
+        case ts.SyntaxKind.PropertySignature:
+          const property = member;
```

Comment:
- `member` is already of type `ts.PropertySignature` because of type narrowing.
- Without type narrowing the following type cast would be needed: `const property = member as ts.PropertySignature;`